tensorflow/compiler/mlir/python/mlir.cc

```diff
@@ -133,6 +148,104 @@ std::string ImportFunction(const std::string &functiondef_proto,
                                  status);
 }
 
+std::string ImportGraphDef(const std::string& proto,
+                           const std::string& pass_pipeline,
+                           bool show_debug_info, TF_Status* status) {
+  GraphDebugInfo debug_info;
+  GraphImportConfig specs;
+  return ImportGraphDefImpl(proto, pass_pipeline, show_debug_info, debug_info,
+                            specs, status);
+}
+
+std::string ImportGraphDef(const std::string& proto,
+                           const std::string& pass_pipeline,
+                           bool show_debug_info, absl::string_view input_names,
+                           absl::string_view input_data_types,
+                           absl::string_view input_data_shapes,
+                           absl::string_view output_names, TF_Status* status) {
+  std::vector<string> node_names = absl::StrSplit(input_names, ',');
+  std::vector<string> node_dtypes = absl::StrSplit(input_data_types, ',');
+  std::vector<string> node_shapes_str = absl::StrSplit(input_data_shapes, ':');
+  std::vector<std::vector<int>> node_shapes;
+
+  std::vector<int> dims;
+  for (const string& shape_str : node_shapes_str) {
+    dims.clear();
+    if(shape_str.empty()) continue;
```

Comment:
Wouldn't skipping empty strings ignore scalar inputs? Or is there a different way to specify the shape for scalars?

Reply:
Indeed can you provide a test? @pashu123 may be able to follow up on this.

Reply:
Setting [this line](https://github.com/tensorflow/tensorflow/pull/49380/files#diff-939c5353442f3db942479a8865c36c9565eee07d60d7157ef35150ff65276562R66) to
```
input_data_shapes=["", ""],
```
results in a failure with
```
tensorflow.python.framework.errors_impl.InvalidArgumentError: Length of input node array and data shape doesn't match
```
This could be used as a test case.

Reply:
@joker-eph Sure I will look into this.

Reply:
Fixed in eaa32809303070b698533a84f6198e0d553ad065